### PR TITLE
Feature/account details

### PR DESCRIPTION
**Pull request by dziennikp**

### CODE REVIEW

File: src/app/user-payment/user-payment.component.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import {filter, map, tap} from 'rxjs/operators';
 import * as firebase from 'firebase';
 import {Router} from '@angular/router';
 import {AuthService} from '../core/auth.service';
-import {ConfigService} from '../config.service';
+// import {ConfigService} from '../config.service';
```
Comment (ammoqq) [inline src/app/user-payment/user-payment.component.ts]: Can be removed?

**Review by ammoqq — APPROVED** [review]

File: src/app/core/user.model.ts [hunk 2]
```diff
@@ -2,10 +2,12 @@ export class FirebaseUserModel {
   image: string;
   name: string;
   provider: string;
+  email: string;
 
   constructor(){
```
Comment (tomson7777) [inline src/app/core/user.model.ts]: why there is constructor? where you create user instance? if you create instance then just user.ts , not model. And remove constructor and you can assign values when you create field.   image = '' etc. But i think it should be just interface, without constructor.

File: src/app/account-details/account-details.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+@Injectable()
+export class AccountDetailsResolver implements Resolve<any> {
+
+  constructor(public userService: UserService, private router: Router) { }
+
+  resolve(route: ActivatedRouteSnapshot): Promise<[any]> {
+    let userDetails: any
+    return new Promise((resolve, reject) => {
+      this.userService.fetchUserData()
```
Comment (tomson7777) [inline src/app/account-details/account-details.resolver.ts]: fetchUserData returns promise, no sense to wrap it in outer promise. Remove this outer and this resolve and reject inside

File: src/app/account-details/account-details.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import { UserService } from '../core/user.service';
+
+
+@Injectable()
+export class AccountDetailsResolver implements Resolve<any> {
+
+  constructor(public userService: UserService, private router: Router) { }
+
+  resolve(route: ActivatedRouteSnapshot): Promise<[any]> {
+    let userDetails: any
```
Comment (tomson7777) [inline src/app/account-details/account-details.resolver.ts]: why any type

File: src/app/core/user.service.ts [hunk 5]
```diff
@@ -26,15 +29,30 @@ export class UserService {
     });
   }
 
-  updateCurrentUser(value){
+  fetchUserData() {
+    return new Promise<any>( (resolve, reject) => {
+      this.database.list<any>(`investors/${firebase.auth().currentUser.uid}`)
+        .valueChanges()
+        .subscribe( submissions => {
```
Comment (tomson7777) [inline src/app/core/user.service.ts:36]: why observable in promise? just return only observable, without subscribe. Subscribe should be in component class. If you want to sort it, just map it for example:

...valueChanges().pipe(
 map(submissions => {
   submissions.sort((n1, n2) => {
            if (n1.createdAt > n2.createdAt) { return 1; }
            if (n1.createdAt < n2.createdAt) { return -1; }
            return 0;
          });
   return submissions[submissions.length - 1]);
}

  )
);

File: src/app/account-details/account-details.component.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    public authService: AuthService,
+    private route: ActivatedRoute,
+    private db: AngularFireDatabase,
+    private storage: AngularFireStorage,
+  ) { }
+
+  ngOnInit() {
+    this.route.data.subscribe(routeData => {
+      const data = routeData['data'];
+      if (data) {
```
Comment (tomson7777) [inline src/app/account-details/account-details.component.ts]: this IF is redundant, you can be sure you get this data. If data is not provided, then you already handled this in resolver, by navigating user back to login (this.router.navigate(['/login'] on error)

File: src/app/app.routes.ts [hunk 7]
```diff
@@ -21,6 +23,7 @@ export const rootRouterConfig: Routes = [
       { path: 'userPayment', component: UserPaymentComponent, canActivate: [AuthGuard]  },
       { path: 'status', component: StatusComponent, canActivate: [AuthGuard],  resolve: { data: StatusResolver} },
```
Comment (tomson7777) [inline src/app/app.routes.ts]: data key in resolve is bad idea, then in the class component you get data.data.  
Replace it with:
resolve: { status: StatusResolver }

File: src/app/app.routes.ts [hunk 8]
```diff
@@ -21,6 +23,7 @@ export const rootRouterConfig: Routes = [
...
       { path: 'create', component: CreateComponent, canActivate: [AuthGuard]  },
+      { path: 'account-details', component: AccountDetailsComponent, canActivate: [AuthGuard], resolve: { data: AccountDetailsResolver}}
```
Comment (tomson7777) [inline src/app/app.routes.ts]: should be resolve: { accountDetails: AccountDetailsResolver}

File: src/app/account-details/account-details.component.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  templateUrl: './account-details.component.html',
+  styleUrls: ['./account-details.component.scss']
+})
+export class AccountDetailsComponent implements OnInit {
+  user: FirebaseUserModel ;
+  userDetails: any;
+  imageBack: string;
+  imageFront: string;
+  constructor(
+    public matDialog: MatDialog,
```
Comment (tomson7777) [inline src/app/account-details/account-details.component.ts]: all services private

File: src/app/account-details/account-details.component.ts [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    this.route.data.subscribe(routeData => {
+      const data = routeData['data'];
+      if (data) {
+        this.userDetails = data;
+        this.onUserFetch();
+      }
+    });
+  }
+
+  onUserFetch() {
```
Comment (tomson7777) [inline src/app/account-details/account-details.component.ts]: method should be private

File: src/app/account-details/account-details.resolver.ts (hunk 3 above)
Comment (dziennikp) [reply src/app/account-details/account-details.resolver.ts]: Not sure how to do it, as this.resolve leads to infinity loop here

**Pull request merged**